docs/rules/rule_md050.md

```diff
@@ -0,0 +1,69 @@
+# # Rule - MD050
+
+| Aliases |
+| --- |
+| `md050` |
+| `refs-regex` |
+
+## Summary
+
+References should match the configured regexes.
+
+## Reasoning
+
+Larger projects or organisations may have rules and requirements regarding the
+format or content of references. The markdown may need to be viewable offline 
+or outside a company network.
+
+## Examples
+
+This rule fails whenever a reference doesn't match one of the configured regexes.
+For instance, a regex that disallows all URIs using HTTP. 
+
+Configuration:
+```json
+"md050": {
+    "enabled": True,
+    "regexes": {
+        "1": {
+            "regex": "^(https?:)?(\/\/)?",
+            "errorMessage": "No external web links",
+        }
+    },
+}
+
+```
+
+
+````Markdown
+[some-image](https//example.com/funny-cat.jpg)
+
+````
+
+## Configuration
+
+| Prefixes |
+| --- |
+| `plugins.md050.` |
+| `plugins.refs-regex.` |
+
+| Value Name | Type | Default | Description |
+| -- | -- | -- | -- |
+| `enabled` | `boolean` | `False` | Whether the plugin rule is enabled. |
```

Comment:
I would recommend to align the table, so that it looks pretty without a markdown viewer.